src/frequenz/sdk/actor/_power_managing/_base_classes.py

```diff
@@ -0,0 +1,189 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Base classes for the power manager."""
+
+from __future__ import annotations
+
+import abc
+import dataclasses
+import datetime
+import enum
+import typing
+
+from ... import timeseries
+from ...timeseries import Power
+
+if typing.TYPE_CHECKING:
+    from ...timeseries.battery_pool import PowerMetrics
+    from .. import power_distributing
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class ReportRequest:
+    """A request to start a reporting stream."""
+
+    source_id: str
+    """The source ID of the actor sending the request."""
+
+    battery_ids: frozenset[int]
+    """The battery IDs to report on."""
+
+    priority: int
+    """The priority of the actor ."""
+
+    def get_channel_name(self) -> str:
+        """Get the channel name for the report request.
+
+        Returns:
+            The channel name to use to identify the corresponding report channel
+                from the channel registry.
+        """
+        return f"power_manager.report.{self.battery_ids=}.{self.priority=}"
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class Report:
+    """Current PowerManager report for a set of batteries."""
+
+    target_power: Power | None
+    """The currently set power for the batteries."""
+
+    inclusion_bounds: timeseries.Bounds[Power] | None
+    """The available inclusion bounds for the batteries, for the actor's priority.
+
+    These bounds are adjusted to any restrictions placed by actors with higher
+    priorities.
+    """
+
+    exclusion_bounds: timeseries.Bounds[Power] | None
+    """The exclusion bounds for the batteries.
+
+    The power manager doesn't manage exclusion bounds, so these are aggregations of
+    values reported by the microgrid API.
+
+    These bounds are adjusted to any restrictions placed by actors with higher
+    priorities.
+    """
+
+    distribution_result: power_distributing.Result | None
+    """The result of the last power distribution.
+
+    This is `None` if no power distribution has been performed yet.
+    """
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class Proposal:
+    """A proposal for a battery to be charged or discharged."""
+
+    source_id: str
+    """The source ID of the actor sending the request."""
+
+    preferred_power: Power | None
+    """The preferred power to be distributed to the batteries.
+
+    If `None`, the preferred power of higher priority actors will get precedence.
+    """
+
+    bounds: timeseries.Bounds[Power | None]
+    """The power bounds for the proposal.
+
+    These bounds will apply to actors with a lower priority, and can be overridden by
+    bounds from actors with a higher priority.  If None, the power bounds will be set to
+    the maximum power of the batteries in the pool.  This is currently and experimental
```

Comment:
```suggestion
    the maximum power of the batteries in the pool.  This is currently an experimental
```

Reply:
I wonder if `bounds` is an experimental feature.  Maybe we should rename it back to `_bounds` in the interface.  But the interface has a new name `propose_power` won't make sense without the bounds.  Is tricky.

Reply:
at least you should replace the and with an an ;)

Reply:
done